onnxruntime/core/providers/cpu/tensor/unsqueeze.cc

```diff
@@ -31,12 +41,26 @@ Status UnsqueezeBase::PrepareCompute(OpKernelContext* ctx, Prepare& p) const {
   ORT_ENFORCE(X != nullptr);
   auto& input_tensor = *X;
 
-  // New dimension count is the current dimensions + the number of entries in axes_
+  std::vector<int64_t> axes;
+  size_t num_inputs = ctx->InputCount();
+  if (num_inputs == 2) { //axes is an input
+    const Tensor* axes_tensor = ctx->Input<Tensor>(1);
+    ORT_ENFORCE(axes_tensor != nullptr, "Axes input is null");
+    ORT_ENFORCE(axes_tensor->Shape().NumDimensions() == 1,
```

Comment:
Can axes be a scalar in which case (NumDimensions() will be 0) - should we enforce Shape().Size() == 1 (which covers scalars) ?

Reply:
going with the form of the original 'axes' attribute, which is a vector, this is not needed. It can be a 1D vector with a single element instead.

Reply:
That may be true, but I ran into a test case (while supporting controlflow ops) wherein a subgraph had an Unsqueeze that took in a scalar for the axes tensor and the spec itself is unclear about the axes needing to be a 1D tensor - https://github.com/onnx/onnx/blob/master/docs/Operators.md#Unsqueeze. Given that the spec is unclear, it might be better to go with the looser check -(check if the number of elements in the tensor is 1 - scalar or otherwise). I l will do it in my change.